libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs

```diff
@@ -0,0 +1,495 @@
+  dRepAcceptedRatio,
+  gsProposalProcedures,
+  spoAcceptedRatio,
+ )
+import Cardano.Ledger.Conway.Tx (AlonzoTx)
+import Cardano.Ledger.Conway.TxCert (ConwayDelegCert, ConwayGovCert, ConwayTxCert)
+import Cardano.Ledger.Credential (Credential)
+import Cardano.Ledger.Crypto (Crypto, StandardCrypto)
+import Cardano.Ledger.Keys (KeyRole (..))
+import Cardano.Ledger.TxIn (TxId)
+import Constrained
+import Control.Monad.Identity (Identity)
+import Data.Bifunctor (Bifunctor (..))
+import Data.Default.Class (Default (..))
+import Data.Foldable (Foldable (..))
+import qualified Data.List.NonEmpty as NE
+import Data.Map.Strict (Map)
+import qualified Data.OSet.Strict as OSet
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import Lens.Micro ((&), (.~), (^.))
+import qualified Lib as Agda
+import Test.Cardano.Ledger.Common (Arbitrary (..))
+import Test.Cardano.Ledger.Conformance (
+  ExecSpecRule (..),
+  OpaqueErrorString (..),
+  SpecTranslate (..),
+  checkConformance,
+  computationResultToEither,
+  runConformance,
+  runSpecTransM,
+ )
+import Test.Cardano.Ledger.Conformance.SpecTranslate.Conway.Base (ConwayExecEnactEnv)
+import Test.Cardano.Ledger.Constrained.Conway (
+  EpochExecEnv,
+  IsConwayUniv,
+  ProposalsSplit,
+  certEnvSpec,
+  certStateSpec,
+  dStateSpec,
+  delegCertSpec,
+  delegEnvSpec,
+  epochEnvSpec,
+  epochSignalSpec,
+  epochStateSpec,
+  genProposalsSplit,
+  govCertEnvSpec,
+  govCertSpec,
+  govEnvSpec,
+  govProceduresSpec,
+  govProposalsSpec,
+  newEpochStateSpec,
+  txCertSpec,
+  utxoEnvSpec,
+  utxoStateSpec,
+  utxoTxSpec,
+  vStateSpec,
+ )
+import Test.Cardano.Ledger.Constrained.Conway.Instances ()
+import Test.Cardano.Ledger.Conway.ImpTest (impAnn, logEntry)
+import Test.Cardano.Ledger.Imp.Common hiding (arbitrary, forAll, prop)
+
+data ConwayGovTransContext era
+  = ConwayGovTransContext
+      (EnactState era)
+      (TxId (EraCrypto era))
+      (Proposals era)
+  deriving (Generic)
+
+deriving instance EraPParams era => Eq (ConwayGovTransContext era)
+
+deriving instance EraPParams era => Show (ConwayGovTransContext era)
+
+instance
+  ( Era era
+  , ToExpr (PParamsHKD Identity era)
+  , ToExpr (PParamsHKD StrictMaybe era)
+  ) =>
+  ToExpr (ConwayGovTransContext era)
+
+instance HasSimpleRep (ConwayGovTransContext era)
+
+instance
+  ( IsConwayUniv fn
+  , EraPParams era
+  , HasSpec fn (EnactState era)
+  , HasSpec fn (Proposals era)
+  ) =>
+  HasSpec fn (ConwayGovTransContext era)
+
+instance Inject (ConwayGovTransContext era) (EnactState era) where
+  inject (ConwayGovTransContext x _ _) = x
+
+instance EraCrypto era ~ c => Inject (ConwayGovTransContext era) (TxId c) where
+  inject (ConwayGovTransContext _ x _) = x
+
+instance EraCrypto era ~ c => Inject (ConwayGovTransContext era) (Proposals era) where
+  inject (ConwayGovTransContext _ _ x) = x
+
+instance
+  ( NFData (SpecRep (ConwayGovPredFailure Conway))
+  , IsConwayUniv fn
+  ) =>
+  ExecSpecRule fn "GOV" Conway
```

Comment:
Is the plan to move each of these ExecSpecRule in their own module?

Reply:
Yes, that's the plan :+1: and also the same for `SpecTranslate`.